modeling/algorithm/algorithms.go

```diff
@@ -23,7 +23,10 @@ func NewAlgorithm(name string) RoutingAlgorithm {
 	switch name {
 	case "SharedGlobal", "SharedGlobalAlgorithm":
 		klog.Info("SharedGlobalAlgorithm created")
-		return SharedGlobalAlgorithm{globalWeight: 0.4, globalThreshold: 100}
+		return SharedGlobalAlgorithm{sharedCoreAlgorithm: SharedGlobalAlgorithmCore{globalWeight: 0.4, globalThreshold: 100}}
+	case "SharedGlobalExclude", "SharedGlobalAlgorithmExclude":
```

Comment:
Maybe this should be called SharedMultiZone since it's no longer necessarily a global slice?